content/en/blog/2023/why-histograms/index.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Why Histograms?
+date: 2023-05-05
+author: '[Daniel Dyla](https://github.com/dyladan)'
+---
+
+<i>Originally posted to
+https://dyladan.me/histograms/2023/05/02/why-histograms/</i>
+
+A histogram is a multi-value counter which summarizes the distribution of one or
+more data points. For example, a histogram may have 3 counters which count the
+occurrences of negative, positive, and zero values respectively. Given a series
+of numbers, `3`, `-9`, `7`, `6`, `0`, and `-1`, the histogram would count `2`
+negative, `1` zero, and `3` positive values. A single histogram data point is
+most commonly represented as a bar chart.
+
+![histogram point as bar chart](hist-point.png "A single histogram point plotted as a bar chart with 3 buckets titled 'Positivity of numbers'. The first bucket shows negative numbers and has a height of 2. The second bucket shows zero values and has a height of 1. The third bucket shows positive values and has a height of 3.")
+
+This simple example has only 3 possible output values, but it is not uncommon to
+have many more in a single histogram. Similarly, this shows only 1 histogram
+data point, but in the world of observability data is usually constantly
+exported. For example, an application may export a histogram every minute which
+summarizes a metric for the previous minute. In this way, you can study how the
+distribution of your data changes over time.
+
+# What are histograms for?
+
+There are many uses for histograms, but their power comes from the ability to
+efficiently answer queries about the distribution of your data. These queries
+most commonly come in some form like "what was the median response time in the
+last minute?" These are known as φ-quantiles, and often are abbreviated in a
+shorthand like `p50` for the 50th percentile or 0.5-quantile, also known as the
+median. More generally, the φ-quantile is the observation value that ranks at
+number φ\*N among the N observations.
+
+# Why is that useful?
+
+The most common use-case for histograms in the observability space is defining
+service level objectives (SLOs). One example of such an SLO might be ">=99% of
+all queries should respond in less than 30ms," or "90% of all page loads should
+become interactive within 100ms of first paint."
```

Comment:
```suggestion
A common use-case for histograms in observability is defining
service level objectives (SLOs). One example of such an SLO might be ">=99% of
all queries should respond in less than 30ms," or "90% of all page loads should
become interactive within 100ms of first paint."
```

Reply:
I don't know if this should be caveated here or not (we'd want to do that in docs though). This point is true when you're working on a metrics data store, but it won't necessarily be true for other tools that let you establish an SLO based on raw data from traces or logs. I'll leave that up to you if you wanna add it or not; I don't think it's urgent to caveat.

Reply:
>This point is true when you're working on a metrics data store, but it won't necessarily be true for other tools that let you establish an SLO based on raw data from traces or logs.

Did you mean to put this on a different line? I'm not sure what claim here might be untrue. It is a common use case for histograms. In any case I like the specific copy edit suggestion you made. "The most" is a bit strong